### PR TITLE
add permission callback

### PR DESCRIPTION
**Pull request by cpalessio96**
### All Submissions:

*   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
*   [x] Does your code has proper inline documentation? 

### Changes proposed in this Pull Request:
moved authentication inside permission callback

### CODE REVIEW

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ includes/Api/ApiAuthUpgradePlan.php  -1
+ includes/Api/ApiLicenseValidation.php  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=6062248&bid=19855672)

File: includes/Api/ApiAuthUpgradePlan.php [hunk 1]
```diff
@@ -37,9 +37,12 @@ public function register_auth_plan_route() {
...
 				'methods'             => 'POST',
-				'callback'            => array( $this, 'auth_and_get_plan' ),
-				'permission_callback' => function() {
-					return current_user_can( 'manage_options' );
+				'callback'            => array( $this, 'return_plan' ),
+				'permission_callback' => function( $request ) {
+					$headers        = $request->get_headers();
+					$jwt            = $headers['jwt'][0];
+					$db_license_key = get_option( 'wubtitle_license_key' );
+					return JWT::decode( $jwt, $db_license_key, array( 'HS256' ) );
```
Comment (HyprCoder) [inline includes/Api/ApiAuthUpgradePlan.php]: Rimuovi il codice duplicato usando una funzione dedicata

**Review by HyprCoder — CHANGES_REQUESTED** [review]

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ includes/Api/ApiAuthUpgradePlan.php  -1
+ includes/Api/ApiLicenseValidation.php  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=6062248&bid=19855672)

File: includes/Api/ApiAuthUpgradePlan.php [hunk 2]
```diff
@@ -37,9 +37,10 @@ public function register_auth_plan_route() {
 			'/auth-plan',
 			array(
 				'methods'             => 'POST',
-				'callback'            => array( $this, 'auth_and_get_plan' ),
-				'permission_callback' => function() {
-					return current_user_can( 'manage_options' );
+				'callback'            => array( $this, 'return_plan' ),
+				'permission_callback' => function( $request ) {
+					$helpers = new Helpers();
```
Comment (HyprCoder) [inline includes/Api/ApiAuthUpgradePlan.php]: create the helper instance in construtor

**Review by HyprCoder — CHANGES_REQUESTED** [review]

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ includes/Api/ApiAuthUpgradePlan.php  -1
+ includes/Api/ApiLicenseValidation.php  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=6062248&bid=19855672)

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ includes/Api/ApiAuthUpgradePlan.php  -1
+ includes/Api/ApiLicenseValidation.php  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=6062248&bid=19855672)

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ includes/Api/ApiStoreSubtitle.php  1
         

Complexity decreasing per file
==============================
+ includes/Api/ApiLicenseValidation.php  -1
+ includes/Api/ApiAuthUpgradePlan.php  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=6062248&bid=19855672)

**Review by HyprCoder — APPROVED** [review]

**Pull request merged**